apps/service_persist/lib/persist/data_file.ex

```diff
@@ -1,17 +1,92 @@
 defmodule Persist.DataFile do
+  use Properties, otp_app: :service_persist
   @type t :: term
-  @type file_path :: String.t()
+  @type file_path :: Path.t()
 
+  @type format :: :json | :orc | :avro
```

Comment:
do you want to include `:avro` if it doesn't support all of the forms of data we want to be able to accept?

Reply:
at the moment, yes